eng/pipelines/templates/jobs/sdk-diff-tests.yml

```diff
@@ -0,0 +1,133 @@
+      echo "##vso[task.setvariable variable=DotnetDotnetBuildId]$dotnet_dotnet_build"
+    displayName: Find associated builds
+    name: Get_Build_Ids
+    env:
+      AZURE_DEVOPS_EXT_PAT: $(System.AccessToken)
+
+  - task: DownloadPipelineArtifact@2
+    displayName: Download MSFT SDK
+    inputs:
+      buildType: specific
+      buildVersionToDownload: specific
+      project: internal
+      pipeline: $(INSTALLER_OFFICIAL_CI_PIPELINE_ID)
+      buildId: $(InstallerBuildId)
+      artifact: BlobArtifacts
+      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-@(alpha|preview|rc|rtm)*)-linux-${{ parameters.architecture }}.tar.gz' 
+      allowPartiallySucceededBuilds: true
+      allowFailedBuilds: true
+      downloadPath: $(Pipeline.Workspace)/Artifacts
+      checkDownloadedFiles: true
+
+  - task: DownloadPipelineArtifact@2
+    displayName: Download Source Build SDK
+    inputs:
+      buildType: specific
+      buildVersionToDownload: specific
+      project: internal
+      pipeline: $(DOTNET_DOTNET_CI_PIPELINE_ID)
+      buildId: $(DotnetDotnetBuildId)
+      artifact: ${{ parameters.buildName }}_${{ parameters.architecture }}_Artifacts
+      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-@(alpha|preview|rc|rtm)*)-${{ parameters.targetRid }}.tar.gz' 
+      allowPartiallySucceededBuilds: true
+      allowFailedBuilds: true
+      downloadPath: $(Pipeline.Workspace)/Artifacts
+      checkDownloadedFiles: true
+    
+  - script: |
+      msft_sdk_tarball_name=$(find "$(Pipeline.Workspace)/Artifacts" -name "dotnet-sdk-*-linux-${{ parameters.architecture }}.tar.gz" -exec basename {} \;)
+
+      if [[ -z "$msft_sdk_tarball_name" ]]; then
+        echo "Microsoft SDK tarball does not exist in '$(Pipeline.Workspace)/Artifacts'. The associated build 'https://dev.azure.com/dnceng/internal/_build/results?buildId=$(InstallerBuildId)&view=results' might have failed."
+        exit 1
+      fi
+
+      sdk_tarball_name=$(find "$(Pipeline.Workspace)/Artifacts" -name "dotnet-sdk-*-${{ parameters.targetRid }}.tar.gz" -exec basename {} \;)
+
+      if [[ -z "$sdk_tarball_name" ]]; then
+        echo "Source-build SDK tarball does not exist in '$(Pipeline.Workspace)/Artifacts'. The associated build 'https://dev.azure.com/dnceng/internal/_build/results?buildId=$(DotnetDotnetBuildId)&view=results' might have failed"
+        exit 1
+      fi
+
+      eng/common/build.sh --projects $(Build.SourcesDirectory)/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/Microsoft.DotNet.SourceBuild.SmokeTests.csproj --restore
+
+      echo "##vso[task.setvariable variable=MsftSdkTarballPath]$(Pipeline.Workspace)/Artifacts/$msft_sdk_tarball_name"
+      echo "##vso[task.setvariable variable=SdkTarballPath]$(Pipeline.Workspace)/Artifacts/$sdk_tarball_name"
+    displayName: Prepare Tests
+    workingDirectory: $(Build.SourcesDirectory)
+
+  - script: >
+      .dotnet/dotnet test

```

Comment:
We have looked into this and there are couple of problems why this won't work without some investment. The main problem is that we don't build the VMR in the SDK diff pipeline.
VMR's `build.sh -run-smoke-test` then counts on the fact the VMR has been built already. For instance, it looks for the PV.props, it expects `NuGet.config` and `global.json` to be updated already and so on.(https://github.com/dotnet/dotnet/blob/main/global.json looks like it's pretty much outdated - maybe we could at least map installer's NuGet.config onto there?)

I understand it would be valuable if we had one way of running the smoke tests that would encapsulate all the env vars etc but the question is how much of an investment that would be and whether it shouldn't be a follow-up.

That being said - what logs are you looking for? We're not really doing anything other than building and running the `SmokeTests.csproj`. Should there be a binlog from that or are there some other logs you are after? The actual file diff is in the exception of the failing test (in AzDO's test pane).

Reply:
In general I find it to be a good practice to always collect a binlog in automation.  When building the smoketests, I think we should be gathing the binlog.

Regarding the diff, we emit an update baseline file and we used to capture this.  This makes updating the baseline much simpler when there are multi fragments that need updating.

Reply:
@MilenaHristova let's add the bl/binlog parameters to the `dotnet test` and `eng/common/build.sh -restore` calls so that they produce a binlog and add a step to upload those as pipeline artifacts. Possibly, `eng/common/build.sh` does this already automatically.

Now to the first part of the question:
Should we change SB's `build.sh` to work on a non-built VMR? It would also probably mean adding `NuGet.config` and mirror `global.json` but I don't understand the effect on SB then.

I'd maybe vote for trying to run the tests using `eng/common/build.sh` now by adding `-test` but @MilenaHristova tried it and it complained there's no `Test` target so we need to figure out how to do that. And later we can think about using SB's `build.sh`?

Reply:
Looks good!

![image](https://user-images.githubusercontent.com/7013027/223998365-85b8fad4-c17e-4094-9eb5-13047b706d6b.png)